src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const cleanedSourceString = sourceString.replace(/\s*;\s*/g, ';').trim();
+  const strArray = cleanedSourceString.split(';');
+  const result = {};
+
+  strArray.forEach((str) => {
+    const [key, value] = str.split(':').map((s) => s.trim());
```

Comment:
There is a potential issue if the `split` operation finds no colon in the string, which will result in `key` being the entire string and `value` being `undefined`. You should check if `value` is `undefined` and handle it accordingly.